doc/contributing/docs/style.rst

```diff
@@ -170,6 +170,14 @@ For this reason, it's best to avoid using them.
 Spelling and punctuation
 ------------------------
 
+Tarantool capitalization
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The word "Tarantool" is capitalized because it's a product name.
+The only context where it can start with a lowercase "t" is code.
```

Comment:
Technically it makes legal to use the lowercased variant in a comment in the code. I would say, when a shell commmand / executable assumed.

Reply:
If the comment becomes part of the documentation (say, via `ldoc`), then this rule is applied. Comments that don't end up in the documentation are covered by the code style guide, not by this guide.